shinylive/_url.py

```diff
@@ -0,0 +1,96 @@
+import base64
+import json
+from pathlib import Path
+from typing import List, Literal, Optional, TypedDict, cast
+
+from lzstring import LZString  # type: ignore[reportMissingTypeStubs]
+
+
+def make_shinylive_url(
+    app: str | Path,
+    files: Optional[tuple[str | Path, ...]] = None,
+    mode: str = "editor",
+    language: Optional[str] = None,
+    no_header: bool = False,
+) -> str:
+    """
+    Generate a URL for a [ShinyLive application](https://shinylive.io).
+
+    Parameters
+    ----------
+    app
+        The main app file of the ShinyLive application. This file should be a Python
+        `app.py` or an R `app.R`, `ui.R`, or `server.R` file. This file will be renamed
+         `app.py` or `app.R` for shinylive, unless it's named `ui.R` or `server.R`.
+    files
+        A tuple of file paths to include in the application. On shinylive, these files
+        will be given stored relative to the main `app` file.
+    mode
+        The mode of the application. Defaults to "editor".
+    language
+        The language of the application. Defaults to None.
+    no_header
+        Whether to include a header. Defaults to False.
```

Comment:
This follows the CLI interface, where `--no-header` disables the header. I'd be happy to reverse this for the Python API and have `header=True` by default.